### DISCUSSION TITLE
is the .env file necessary?

### DISCUSSION THREAD

**Discussion by crispybegs** [hargata/lubelog#316] [Q&A] [answered]
i have a habit (for better or for worse) of just throwing entire compose files as they are into portainer stacks, rather than saving separate compose / env yamls and then uploading them.

is it possible to simple copy and paste the content of the .env straight into the environment section of the main compse yaml and use it that way, or is that verboten?

**Comment by joshtbernstein** [comment, marked as answer]
That's what I did. I think it mainly comes down to the SMTP related ones and if you want those credentials in your compose file. I'm not using that feature right now, but if/when, I'll move those variables back into my .env file.
